src/include/parser/table_ref.h

```diff
@@ -31,48 +31,48 @@ class JoinDefinition;
 struct TableRef {
   TableRef(TableReferenceType type)
       : type(type),
-        schema(NULL),
-        table_info_(NULL),
-        alias(NULL),
-        select(NULL),
-        list(NULL),
-        join(NULL) {}
+        schema(nullptr),
+        table_info_(nullptr),
+        alias(nullptr),
+        select(nullptr),
+        list(nullptr),
+        join(nullptr) {}
 
   virtual ~TableRef();
 
   TableReferenceType type;
 
-  char* schema;
+  std::unique_ptr<char[]> schema;
 
   // Expression of database name and table name
-  TableInfo* table_info_ = nullptr;
+  std::unique_ptr<TableInfo> table_info_ = nullptr;
 
-  char* alias;
+  std::unique_ptr<char[]> alias;
 
   SelectStatement* select;
-  std::vector<TableRef*>* list;
-  JoinDefinition* join;
+  std::unique_ptr<std::vector<std::unique_ptr<TableRef>>> list;
+  std::unique_ptr<JoinDefinition> join;
 
   // Convenience accessor methods
-  inline bool HasSchema() { return schema != NULL; }
+  inline bool HasSchema() { return schema != nullptr; }
 
   // Get the name of the database of this table
   inline const char* GetDatabaseName() const {
```

Comment:
Returning a raw pointer defeats the purpose of using smart pointers inside of the classes.